src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -5707,12 +5702,47 @@ public override IOperation VisitDefaultValue(IDefaultValueOperation operation, i
 
         public override IOperation VisitTypeParameterObjectCreation(ITypeParameterObjectCreationOperation operation, int? captureIdForResult)
         {
-            return new TypeParameterObjectCreationExpression(Visit(operation.Initializer), semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, IsImplicit(operation));
+            return new TypeParameterObjectCreationExpression(initializer: null, // PROTOTYPE(dataflow): Dropping initializer for now to enable test hook verification
+                                                             semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, IsImplicit(operation));
         }
 
         public override IOperation VisitInvalid(IInvalidOperation operation, int? captureIdForResult)
         {
-            return new InvalidOperation(VisitArray(operation.Children.ToImmutableArray()), semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, IsImplicit(operation));
+            var children = ArrayBuilder<IOperation>.GetInstance();
+            children.AddRange(operation.Children);
+
+            if (children.Count != 0 && children.Last().Kind == OperationKind.ObjectOrCollectionInitializer)
+            {
+                // We are dealing with erroneous object creation. All children, but the last one are arguments for the constructor,
```

Comment:
> We are dealing with erroneous object creation. All children, but the last one are arguments for the constructor, [](start = 18, length = 113)

I feel uncomfortable with this reverse engineering (inferring from the shape of the invalid operation how it was produced and what it means). 
Is there a reason we can't produce an object creation (with some bits missing) even when overload resolution failed?

Reply:
> I feel uncomfortable with this reverse engineering (inferring from the shape of the invalid operation how it was produced and what it means). 
Is there a reason we can't produce an object creation (with some bits missing) even when overload resolution failed?

This is definitely an error case (an invalid operation), we just need to have reasonable recovery story. ObjectOrCollectionInitializer cannot appear in any context other than and attempt to create an initialize an instance. Ignoring any possible bugs in the compiler, we are pretty safe to make that assumption. We are not creating object creation in the case of overload resolution because we cannot provide symbol for constructor and reliable mapping from arguments to parameters, default values and paramarrays.

---
In reply to: [192865269](https://github.com/dotnet/roslyn/pull/27393#discussion_r192865269) [](ancestors = 192865269)